src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,51 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.exception.InvalidDataException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_AGE = 18;
+    private static final int MIN_LENGTH = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+
+        if (user == null) {
+            throw new InvalidDataException("User cannot be null");
+        }
+        if (Storage.people.contains(storageDao.get(user.getLogin()))) {
+            throw new InvalidDataException("User already exists "
+                    + user.getLogin()
+                    + " with such login");
+        }
```

Comment:
The check `Storage.people.contains(storageDao.get(user.getLogin()))` might not work as expected if `storageDao.get(user.getLogin())` returns null. Consider checking if the user exists in a more reliable way, such as checking if the result of `storageDao.get(user.getLogin())` is not null.